query-node/mappings/giza/storage/index.ts

```diff
@@ -0,0 +1,636 @@
+/*
+eslint-disable @typescript-eslint/naming-convention
+*/
+import { DatabaseManager, EventContext, StoreContext } from '@joystream/hydra-common'
+import { Storage } from '../../generated/types/storage'
+import {
+  DistributionBucket,
+  DistributionBucketFamily,
+  DistributionBucketOperator,
+  DistributionBucketOperatorMetadata,
+  DistributionBucketOperatorStatus,
+  NodeLocationMetadata,
+  StorageBag,
+  StorageBagOwner,
+  StorageBagOwnerChannel,
+  StorageBagOwnerCouncil,
+  StorageBagOwnerMember,
+  StorageBagOwnerWorkingGroup,
+  StorageBucket,
+  StorageBucketOperatorStatusActive,
+  StorageBucketOperatorStatusInvited,
+  StorageBucketOperatorStatusMissing,
+  StorageDataObject,
+  StorageSystemParameters,
+  GeoCoordinates,
+  StorageBagDistributionAssignment,
+  StorageBagStorageAssignment,
+} from 'query-node/dist/model'
+import BN from 'bn.js'
+import { getById, getWorkingGroupModuleName, bytesToString } from '../common'
+import { BTreeSet } from '@polkadot/types'
+import { DataObjectCreationParameters } from '@joystream/types/storage'
+import { registry } from '@joystream/types'
+import { In } from 'typeorm'
+import _ from 'lodash'
+import { DataObjectId, BagId, DynamicBagId, StaticBagId } from '@joystream/types/augment/all'
+import {
+  processDistributionBucketFamilyMetadata,
+  processDistributionOperatorMetadata,
+  processStorageOperatorMetadata,
+} from './metadata'
+
+async function getDataObjectsInBag(
+  store: DatabaseManager,
+  bagId: BagId,
+  dataObjectIds: BTreeSet<DataObjectId>
+): Promise<StorageDataObject[]> {
+  const dataObjects = await store.getMany(StorageDataObject, {
+    where: {
+      id: In(Array.from(dataObjectIds).map((id) => id.toString())),
+      storageBag: { id: getBagId(bagId) },
+    },
+  })
+  if (dataObjects.length !== Array.from(dataObjectIds).length) {
+    throw new Error(
+      `Missing data objects: ${_.difference(
+        Array.from(dataObjectIds).map((id) => id.toString()),
+        dataObjects.map((o) => o.id)
+      )} in bag ${getBagId(bagId)}`
+    )
+  }
+  return dataObjects
+}
+
+function getStaticBagOwner(bagId: StaticBagId): typeof StorageBagOwner {
+  if (bagId.isCouncil) {
+    return new StorageBagOwnerCouncil()
+  } else if (bagId.isWorkingGroup) {
+    const owner = new StorageBagOwnerWorkingGroup()
+    owner.workingGroupId = getWorkingGroupModuleName(bagId.asWorkingGroup)
+    return owner
+  } else {
+    throw new Error(`Unexpected static bag type: ${bagId.type}`)
+  }
+}
+
+function getDynamicBagOwner(bagId: DynamicBagId) {
+  if (bagId.isChannel) {
+    const owner = new StorageBagOwnerChannel()
+    owner.channelId = bagId.asChannel.toNumber()
+    return owner
+  } else if (bagId.isMember) {
+    const owner = new StorageBagOwnerMember()
+    owner.memberId = bagId.asMember.toNumber()
+    return owner
+  } else {
+    throw new Error(`Unexpected dynamic bag type: ${bagId.type}`)
+  }
+}
+
```

Comment:
Shouldn't we reuse the format for the `bagId` from the storage and distribution nodes (`static:council`)?

Reply:
That's a good idea, I think I just wasn't aware of this format when I was creating the mappings.

Reply:
Done in https://github.com/Joystream/joystream/pull/2515/commits/3f04581173a720ac3a15a10c673d19875e0b4961